### PR TITLE
RHDEVDOCS-3355: Document the sizing requirements for GitOps Operator and the default instance

### PR DESCRIPTION
**Pull request by HarshCasper**
- JIRA ID: [RHDEVDOCS-3355](https://issues.redhat.com/browse/RHDEVDOCS-3355)
- Aligned team: GitOps
- SME review: 
- Peer Review:  
- Docs Preview: https://deploy-preview-42876--osdocs.netlify.app/openshift-enterprise/latest/cicd/gitops/about-sizing-requirements-gitops

### CODE REVIEW

**Review by jannfis — APPROVED** [review]
/lgtm

File: modules/sizing-requirements-for-gitops.adoc [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+:_content-type: CONCEPT
+[id="sizing-requirements-for-gitops_{context}"]
+= Sizing requirements for GitOps
+
+Red Hat Openshift GitOps is a declarative way to implement continuous deployment for cloud native applications. Through GitOps, you can define and configure the CPU and Memory requirements of your application. Everytime you install the GitOps operator, the resources on the namespace is installed within the defined limits. If the default install does not set any limits or requests, the operator will fail within the namespace which has quotas. Without enough resources, the cluster will not be able to schedule ArgoCD related pods. The resource requests and limits for the default workloads are:
+
+[cols=",,,,",options="header",]
+|===
+|Workload |CPU requests |CPU limits |Mem requests |Mem limits
```
[the remark this answers is not in the document]
Comment (Srivaralakshmi) [reply modules/sizing-requirements-for-gitops.adoc:13]: Suggestion: try reducing the width of columns 2,3, 4, and 5?

File: modules/sizing-requirements-for-gitops.adoc [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+|argocd-application-controller |1 |2 |1024M |2048M
+|applicationset-controller |1 |2 |512M |1024M
+|argocd-server |0.125 |0.5 |128M |256M
+|argocd-repo-server |0.5 |1 |256M |1024M
+|argocd-redis |0.25 |0.5 |128M |256M
+|argocd-dex |0.25 |0.5 |128M |256M
+|HAProxy |0.25 |0.5 |128M |256M
+|===
+
+Optionally, you can also use the ArgoCD custom resource with `kubectl`/`oc` command to see the specifics and modify them:
```
Comment (Srivaralakshmi) [inline modules/sizing-requirements-for-gitops.adoc:23]:
```suggestion
Optionally, you can also use the ArgoCD custom resource with the `kubectl` or `oc` command to see the specifics and modify them:
```

**Question:** When using OCP we always replace kubectl with oc, right? Please check and retain only oc if possible. Refer to the OCP documentation and do the needful as appropriate.

**Review by Srivaralakshmi — CHANGES_REQUESTED** [review]
@HarshCasper Left some suggestions. PTAL. Thanks!

File: modules/sizing-requirements-for-gitops.adoc [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+:_content-type: CONCEPT
+[id="sizing-requirements-for-gitops_{context}"]
+= Sizing requirements for GitOps
+
+[role="_abstract"]
+{gitops-title} is a declarative way to implement continuous deployment for cloud-native applications. Through GitOps, you can define and configure the CPU and memory requirements of your application. Every time you install the {gitops-title} Operator, the resources on the namespace are installed within the defined limits. If the default install does not set any limits or requests, the Operator fails within the namespace which has quotas. Without enough resources, the cluster is unable to schedule ArgoCD related pods. The following table details the resource requests and limits for the default workloads:
+
+[cols="2,3,4,5,",options="header",]
+|===
+|Workload |CPU requests |CPU limits |Memory requests |Memory limits
```
Comment (Srivaralakshmi) [inline modules/sizing-requirements-for-gitops.adoc:14]: Suggestion: try reducing the width of columns if possible so that they all look proportionate.

**Review by Srivaralakshmi — APPROVED** [review]
@HarshCasper LGTM

File: _topic_maps/_topic_map.yml [hunk 4]
```diff
@@ -1591,6 +1591,8 @@ Topics:
     File: configuring-sso-for-argo-cd-using-keycloak
   - Name: Running Control Plane Workloads on Infra nodes
     File: run-gitops-control-plane-workload-on-infra-nodes
+  - Name: Sizing requirements for GitOps Operator
```
Comment (rolfedh) [inline _topic_maps/_topic_map.yml:1594]: Good work, capitalizing Operator!

File: cicd/gitops/about-sizing-requirements-gitops.adoc [hunk 5]
```diff
@@ -0,0 +1,12 @@
+:_content-type: ASSEMBLY
+[id="about-sizing-requirements-gitops"]
+= Sizing requirements for GitOps Operator
+include::_attributes/common-attributes.adoc[]
+:context: about-sizing-requirements-gitops
+
+toc::[]
+
+[role="_abstract"]
+The sizing requirements page displays the document sizing requirements for installing {gitops-title} on {product-title}.  It also details the default ArgoCD instance which gets instantiated on the installation of the Operator.
```
Comment (rolfedh) [inline cicd/gitops/about-sizing-requirements-gitops.adoc:10]:
```suggestion
The sizing requirements page displays the document sizing requirements for installing {gitops-title} on {product-title}. It also details the default ArgoCD instance which gets instantiated on the installation of the Operator.
```

Comment (rolfedh) [reply cicd/gitops/about-sizing-requirements-gitops.adoc:10]:
```suggestion
The sizing requirements page displays the document sizing requirements for installing {gitops-title} on {product-title}.  It also details the default ArgoCD instance, which gets instantiated upon the installation of the Operator.
```

Comment (rolfedh) [reply cicd/gitops/about-sizing-requirements-gitops.adoc:10]:
```suggestion
The sizing requirements page displays the document sizing requirements for installing {gitops-title} on {product-title}.  It also details the default ArgoCD instance that installing the Operator instantiates.
```

File: modules/sizing-requirements-for-gitops.adoc [hunk 6]
```diff
@@ -0,0 +1,29 @@
+// Module is included in the following assemblies:
+//
+// * openshift-docs/cicd/gitops/about-sizing-requirements-gitops.adoc
+
+:_content-type: CONCEPT
+[id="sizing-requirements-for-gitops_{context}"]
+= Sizing requirements for GitOps
+
+[role="_abstract"]
+{gitops-title} is a declarative way to implement continuous deployment for cloud-native applications. Through GitOps, you can define and configure the CPU and memory requirements of your application. Every time you install the {gitops-title} Operator, the resources on the namespace are installed within the defined limits. If the default install does not set any limits or requests, the Operator fails within the namespace which has quotas. Without enough resources, the cluster is unable to schedule ArgoCD related pods. The following table details the resource requests and limits for the default workloads:
```
Comment (rolfedh) [inline modules/sizing-requirements-for-gitops.adoc:10]: Would it help to separate the abstract from the longer explanation?

Comment (rolfedh) [reply modules/sizing-requirements-for-gitops.adoc:10]:
```suggestion
{gitops-title} is a declarative way to implement continuous deployment for cloud-native applications. Through GitOps, you can define and configure the CPU and memory requirements of your application.

 Every time you install the {gitops-title} Operator, the resources on the namespace are installed within the defined limits. If the default install does not set any limits or requests, the Operator fails within the namespace which has quotas. Without enough resources, the cluster is unable to schedule ArgoCD related pods. The following table details the resource requests and limits for the default workloads:
```

Comment (rolfedh) [reply modules/sizing-requirements-for-gitops.adoc:10]: In light of the IBM SG guidance in the "Relative Pronouns" section, would it make sense to review your usage of "which" and "that" throughout this PR?

Comment (rolfedh) [reply modules/sizing-requirements-for-gitops.adoc:10]: `is unable to` > `cannot`

**Review by rolfedh — COMMENTED** [review]
Very nice work overall. I left some minor comments.

File: modules/sizing-requirements-for-gitops.adoc [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+// * openshift-docs/cicd/gitops/about-sizing-requirements-gitops.adoc
+
+:_content-type: CONCEPT
+[id="sizing-requirements-for-gitops_{context}"]
+= Sizing requirements for GitOps
+
+[role="_abstract"]
+{gitops-title} is a declarative way to implement continuous deployment for cloud-native applications. Through GitOps, you can define and configure the CPU and memory requirements of your application.
+
+Every time you install the {gitops-title} Operator, the resources on the namespace are installed within the defined limits. If the default installation does not set any limits or requests, the Operator fails within the namespace which has quotas. Without enough resources, the cluster cannot schedule ArgoCD related pods. The following table details the resource requests and limits for the default workloads:
```
Comment (rolfedh) [inline modules/sizing-requirements-for-gitops.adoc:12]:
```suggestion
Every time you install the {gitops-title} Operator, the resources on the namespace are installed within the defined limits. If the default installation does not set any limits or requests, the Operator fails within the namespace, which has quotas. Without enough resources, the cluster cannot schedule ArgoCD related pods. The following table details the resource requests and limits for the default workloads:
```

**Review by rolfedh — APPROVED** [review]

**Review by rolfedh — COMMENTED** [review]
Please review IBM SG section on "Relative Pronouns" and fix the usage of "which."

File: modules/sizing-requirements-for-gitops.adoc (hunk 6 above)
Comment (rolfedh) [reply modules/sizing-requirements-for-gitops.adoc]: > In light of the IBM SG guidance in the "Relative Pronouns" section, would it make sense to review your usage of "which" and "that" throughout this PR?

This issue has not been resolved.

File: modules/sizing-requirements-for-gitops.adoc (hunk 7 above)
Comment (rolfedh) [inline modules/sizing-requirements-for-gitops.adoc:12]: Again, please review IBM SG section on "Relative Pronouns" and fix this incorrect usage of "which."

**Review by rolfedh — CHANGES_REQUESTED** [review]
Please see open issue about "which."

File: cicd/gitops/about-sizing-requirements-gitops.adoc [hunk 8]
```diff
@@ -0,0 +1,12 @@
+:_content-type: ASSEMBLY
+[id="about-sizing-requirements-gitops"]
+= Sizing requirements for GitOps Operator
+include::_attributes/common-attributes.adoc[]
+:context: about-sizing-requirements-gitops
+
+toc::[]
+
+[role="_abstract"]
+The sizing requirements page displays the document sizing requirements for installing {gitops-title} on {product-title}. It also details the default ArgoCD instance that installing the Operator instantiates.
```
[the remark this answers is not in the document]
Comment (Preeticp) [reply cicd/gitops/about-sizing-requirements-gitops.adoc:10]: It also details the default ArgoCD instance that installing the Operator instantiates.

Suggest: It also provides the sizing details for the default ArgoCD instance that is instantiated by the GitOps Operator.

File: modules/sizing-requirements-for-gitops.adoc (hunk 3 above)
Comment (Preeticp) [reply modules/sizing-requirements-for-gitops.adoc:16]: +1 @HarshCasper the first and the last column look very squeezed in comparison to the middle three.

**Comment by Preeticp** [comment]
@HarshCasper I amnot sure what the issue is but we will need to do a manual CP for 4.7, 4.8, 4.9 and 4.10
I have merged the CP PR for 4.11

**Pull request merged**